src/argilla_sdk/records/_dataset_records.py

```diff
@@ -319,17 +325,12 @@ def to_json(self, path: Union[Path, str]) -> Path:
             The path to the file where the records were saved.
 
         """
-        if isinstance(path, str):
-            path = Path(path)
-        if path.exists():
-            raise FileExistsError(f"File {path} already exists.")
-        record_dicts = self(with_suggestions=True, with_responses=True).to_list()
-        with open(path, "w") as f:
-            json.dump(record_dicts, f)
-        return path
+        records = list(self(with_suggestions=True, with_responses=True))
+        return JsonIO.to_json(records=records, path=path)
```

Comment:
why a path and not a json string or anything? not sure what similar behavior is for other packages?

Reply:
The path is not a change from this PR.

There's an issue here : https://github.com/argilla-io/argilla-python/issues/214

That said, the choice of path isn't obvious. We have a number of methods across `Dataset`, `Settings`, and `DatasetRecords` that deal with persistence and serialization. We should unify them so users know how to save their datasets.

In most cases, they can just use:

```python
ds.to_disk("mydir/")
```

Which calls the underlying methods like this one.